### PR TITLE
[REF] use Auth0 for authentication

### LINKED ISSUE #59: NeuroStuff Auth service
Related to #56, let's look into creating a fully fledged auth service. 
https://supertokens.io/ looks promising

### PR DESCRIPTION
**Pull request by jdkent**
closes #59 

Use auth0 for authentication

### CODE REVIEW

File: neurostore/models/auth.py [hunk 1]
```diff
@@ -25,25 +22,11 @@ class Role(BaseMixin, db.Model, RoleMixin):
...
-
-
-class OAuth(OAuthConsumerMixin, db.Model):
-    __tablename__ = "oauth"
-
-    user_id = db.Column(db.Text, db.ForeignKey("users.id"))
-    user = relationship(User, backref=backref("oauth"))
-    provider_user_id = db.Column(db.Text, unique=True, nullable=False)
-    provider = db.Column(db.Text)
+    neuroid = db.Column(db.Text, unique=True)
```
Comment (adelavega) [inline neurostore/models/auth.py:27]:
```suggestion
    external_id = db.Column(db.Text, unique=True)
```

File: neurostore/resources/users.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+class UserView(ObjectView):
+    _model = User
+
+    @property
+    def schema(self):
+        return globals()[self._model.__name__ + 'Schema']
+
+    def put(self, id):
+        current_user = User.query.filter_by(neuroid=connexion.context['user']).first()
```
Comment (adelavega) [inline neurostore/resources/users.py:44]:
```suggestion
        current_user = User.query.filter_by(external_id=connexion.context['user']).first()
```

I'm sure I  missed a few more instance of these

File: neurostore/schemas/auth.py [hunk 3]
```diff
@@ -1,33 +1,14 @@
...
-        if len(value) < 6:
-            raise ValidationError('Password must be at least 6 characters.')
-
-    @post_load()
-    def encrypt_password(self, in_data, **kwargs):
-        if 'password' in in_data:
-            in_data['password'] = encrypt_password(in_data['password'])
-        return in_data
+    name = fields.Str(description='User full name')
+    neuroid = fields.Str(description="universal id")
```
Comment (adelavega) [inline neurostore/schemas/auth.py:11]:
```suggestion
    external_id = fields.Str(description="External authentication service user ID")
```

**Review by adelavega — CHANGES_REQUESTED** [review]
Looks good, just some very minor suggestions

File: neurostore/tests/api/test_users.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+def test_create_user(auth_client):
+    new_user = {
+        'name': "fake name",
+        'neuroid': '1234',
+    }
+    resp = auth_client.post("/api/users/", data=new_user)
+
+    assert resp.status_code == 200
+    assert User.query.filter_by(neuroid='1234').first() is not None
```
Comment (jdkent) [inline neurostore/tests/api/test_users.py:12]:
```suggestion
    assert User.query.filter_by(external_id='1234').first() is not None
```

File: neurostore/tests/conftest.py [hunk 5]
```diff
@@ -89,21 +93,49 @@ def auth_client(add_users):
...
+            username=name + "@email.com",
+            password=passw,
+            realm='Username-Password-Authentication',
+            audience=app.config['AUTH0_API_AUDIENCE'],
+            scope='openid',
+        )
+        token_info = decode_token(payload['access_token'])
+        user_datastore.create_user(
+            name=name,
+            neuroid=token_info['sub'],
```
Comment (jdkent) [inline neurostore/tests/conftest.py:130]:
```suggestion
            external_id=token_info['sub'],
```

File: neurostore/tests/conftest.py [hunk 6]
```diff
@@ -89,21 +93,49 @@ def auth_client(add_users):
...
+        )
+
+        tokens[name] = {
+            'token': payload['access_token'],
+            'id': user_datastore.find_user(neuroid=token_info['sub']).id,
```
Comment (jdkent) [inline neurostore/tests/conftest.py:135]:
```suggestion
            'id': user_datastore.find_user(external_id=token_info['sub']).id,
```

File: neurostore/tests/api/test_users.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
+from ...models import User
+
+
+def test_create_user(auth_client):
+    new_user = {
+        'name': "fake name",
+        'neuroid': '1234',
```
Comment (jdkent) [inline neurostore/tests/api/test_users.py:7]:
```suggestion
        'external_id': '1234',
```

**Pull request merged**